### PR TITLE
feat(api): add alpha types package; custom id; custom sel set support

### PR DESCRIPTION
**Pull request by iartemiev**
#### Description of changes

Increased bundle size is from upgrading `tslib@1.x` -> `2.6`

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-v6-types-package/index.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+					: never;
+			} & {
+				[k in Paths[number] as k extends `${string}.${string}`
+					? never
+					: k]: M[k];
+			}
+	  >;
+
+type ModelIdentifier<Model extends Record<any, any>> = Prettify<
+	Record<Model['identifier'] & string, string>
```
Comment (manueliglesias) [inline packages/amplify-v6-types-package/index.ts:51]: Identifier fields can be other types than string (most of the time `number`), is is possible to adjust this type to account for that?

File: packages/api/package.json [hunk 2]
```diff
@@ -26,16 +26,16 @@
...
 		"test:size": "size-limit",
 		"build-with-test": "npm test && npm run build",
-		"build:cjs": "node ./build es5 && webpack && webpack --config ./webpack.config.dev.js",
-		"build:esm": "node ./build es6",
-		"build:cjs:watch": "node ./build es5 --watch",
-		"build:esm:watch": "node ./build es6 --watch",
+		"build:cjs": "rimraf lib && tsc -p ./tsconfig.build.json -m commonjs --outDir lib && webpack && webpack --config ./webpack.config.dev.js",
+		"build:esm": "rimraf lib-esm && tsc -p ./tsconfig.build.json -m esnext --outDir lib-esm",
+		"build:cjs:watch": "rimraf lib && tsc -m commonjs --outDir lib --watch",
+		"build:esm:watch": "rimraf lib-esm && tsc -m esnext --outDir lib-esm --watch",
```
Comment (manueliglesias) [inline packages/api/package.json:32]: For the sake of consistency, can we keep using the build script? (and made changes to it if needed?)

File: packages/api/src/API.ts [hunk 3]
```diff
@@ -83,6 +84,11 @@ export class APIClass extends InternalAPIClass {
 						// flatten response
 						if (res.data !== undefined) {
 							const [key] = Object.keys(res.data);
+
+							if (res.data[key].items) {
+								return res.data[key].items;
+							}
+
 							return res.data[key];
 						}
```
Comment (manueliglesias) [inline packages/api/src/API.ts:93]: It would be better to use the information coming from the model instrospection

**Review by manueliglesias — APPROVED** [review]
Non blocking comments only

LGTM!

File: packages/api/package.json (hunk 2 above)
Comment (iartemiev) [reply packages/api/package.json:32]: Nope, we can't use the build script if we want to use a newer version of TS. Have to use this approach but it's aligned with what we do in Core for the same purpose

File: packages/amplify-v6-types-package/index.ts (hunk 1 above)
Comment (iartemiev) [reply packages/amplify-v6-types-package/index.ts:51]: Yes, I'm going to circle back and improve this type in the next PR

File: packages/api/src/API.ts (hunk 3 above)
Comment (iartemiev) [reply packages/api/src/API.ts:93]: For this one, we can probably just narrow it based on the operation, e.g. `list` => flatten items. I'm not sure model intro will give us the necessary metadata

**Pull request merged**